core/src/main/java/io/parsingdata/metal/expression/value/reference/Last.java

```diff
@@ -49,4 +50,16 @@ public String toString() {
         return getClass().getSimpleName() + "(" + operand + ")";
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        return obj != null
+            && getClass() == obj.getClass()
+            && Objects.equals(operand, ((Last)obj).operand);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(operand);
```

Comment:
A few of these `hashCode()` implementations are exactly the same. For example: `First` and `Last`. Could that pose a problem?